### PR TITLE
implement settings screen

### PR DESCRIPTION
**Pull request by TheRemakerMan**
also added prefs object class.

### CODE REVIEW

File: app/src/main/java/org/cosmicide/rewrite/extension/editor.kt [hunk 1]
```diff
@@ -24,6 +24,6 @@ fun CodeEditor.setLanguageTheme(language: EditorLanguage) {
  * Sets the font and enables highlighting of the current line for the code editor.
  */
 fun CodeEditor.setFont() {
-    typefaceText = ResourcesCompat.getCachedFont(context, R.font.noto_sans_mono)
+    typefaceText = ResourcesCompat.getFont(context, R.font.noto_sans_mono)
```
Comment (PranavPurwar) [inline app/src/main/java/org/cosmicide/rewrite/extension/editor.kt:27]: getCachedFont would be better here i think

Comment (TheRemakerMan) [reply app/src/main/java/org/cosmicide/rewrite/extension/editor.kt:27]: With cachedfont, the font is not loaded.

Comment (TheRemakerMan) [reply app/src/main/java/org/cosmicide/rewrite/extension/editor.kt:27]: With cachedfont, the font is not loaded.

File: app/src/main/res/values/arrays.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string-array name="java_version_entries">
+        <item>8</item>
+        <item>11</item>
```
Comment (PranavPurwar) [inline app/src/main/res/values/arrays.xml:5]: what about other version? For ex if you just want to use text blocks and want compatibility with Java 16, you can just use it

Comment (TheRemakerMan) [reply app/src/main/res/values/arrays.xml:5]: okay, i will add every java version from 8 to 17.

File: app/src/main/java/org/cosmicide/rewrite/extension/editor.kt (hunk 1 above)
Comment (PranavPurwar) [reply app/src/main/java/org/cosmicide/rewrite/extension/editor.kt:27]: It checks if its cached and sets it. atleast in emulator, it loads properly

Comment (TheRemakerMan) [reply app/src/main/java/org/cosmicide/rewrite/extension/editor.kt:27]: You check? With the `getCachedFont()` method on my device, the font just didn't load.

Comment (PranavPurwar) [reply app/src/main/java/org/cosmicide/rewrite/extension/editor.kt:27]: it did work

Comment (TheRemakerMan) [reply app/src/main/java/org/cosmicide/rewrite/extension/editor.kt:27]: Check last artifact (not pr's artifact)

**Pull request merged**